lib/list_validator.rb

```diff
@@ -0,0 +1,11 @@
+class ListValidator
+  def initialize(list)
+    @list = list
```

Comment:
I buy your idea.. will research it a bit.. howbeit this may not be handle huge lists